lib/chef/knife/ec2_server_create.rb

```diff
@@ -666,34 +651,34 @@ def plugin_validate_options!
           exit 1
         end
 
-        if config_value(:ebs_encrypted)
+        if config[:ebs_encrypted]
           error_message = ""
           errors = []
           # validation for flavor and ebs_encrypted
-          if !config_value(:flavor)
+          if !config[:flavor]
             ui.error("--ebs-encrypted option requires valid flavor to be specified.")
             exit 1
-          elsif config_value(:ebs_encrypted) && ! %w{m3.medium m3.large m3.xlarge m3.2xlarge m4.large m4.xlarge
+          elsif config[:ebs_encrypted] && ! %w{m3.medium m3.large m3.xlarge m3.2xlarge m4.large m4.xlarge
```

Comment:
wtf is this mess

Reply:
that's a good question

Reply:
that's just a whitelist of every single AWS instance type that supports "ebs encrypted"